docs/proposals/build-skipping.adoc

```diff
@@ -0,0 +1,73 @@
+= Proposal: Multi build repo skipping
+
+== Purpose
+
+Skip builds for which nothing has changed in their working directory.
+
+== Background
+
+For a repo with a single build, the pipeline build will only be triggered if any changes trigger a new pipeline run. 
+
+However if one has multiple builds in one repo, the pipeline build runs all build tasks even for those areas where nothing has changed. 
+
+A simple scenario is a repo with a backend and frontend component whose build outputs are combined by a subsequent docker build in a subsequent deploy task. 
+
+There is no good reason when only the frontend code changes that the backend build is redone as well and vice versa of course.  
+
+== Solution
+
+The solution is only provided for tasks supporting global caching (see separate proposal # 412).
```

Comment:
I think this is unrelated to "dependency caching" ... so wouldn't that meant that all build tasks support "output dir caching"?